MapboxCoreNavigation/RouteController.swift

```diff
@@ -356,14 +356,22 @@ extension RouteController: CLLocationManagerDelegate {
             firstWaypoint.heading = location.course
             firstWaypoint.headingAccuracy = 90
         }
-        
+
         routeTask = directions.calculate(options, completionHandler: { [weak self] (waypoints, routes, error) in
             guard let strongSelf = self else {
                 return
             }
             
             if let route = routes?.first {
-                strongSelf.routeProgress = RouteProgress(route: route)
+                
+                // If the new route has a first step that has a distance greater than 1km,
+                // surpress the first announcement since they have ample time until the next maneuver
+                var newAlertLevel: AlertLevel = .none
+                if let firstLeg = route.legs.first, let firstStep = firstLeg.steps.first, firstStep.distance > 1_000 {
```

Comment:
The comment above says this heuristic is based on duration, whereas this statement checks only for distance. If the user is traveling along a freeway in the U.S. and the new upcoming maneuver is 1 kilometer ahead, then the user has already passed the standard 1-mile advance guide sign and the next guide sign they’ll see is directly above the exit. Either we’d need to switch to a heuristic based on remaining duration, or we’d need separate logic for freeways versus surface streets.